### ISSUE TITLE
Make boards store tile's locations rather than tiles storing their location

### ISSUE THREAD

**Issue by lead-y** [cdeevfrr/labyrinth#39]

**Comment by lead-y** [comment]
Once problem with this is that it would make it slower to get a location given a tile. We could consider using a BidiMap to avoid this problem.

**Comment by lead-y** [comment]
Actually, we could also just start referring to all tiles by "tileAt()".

**Comment by cdeevfrr** [comment]
Good point lindsay! The difficulty in getting location from tile might mean we need to change how we push tiles as well, since currently tile pushing is done recursively. It's not certain that this is a change we want to make.

**Comment by cdeevfrr** [comment]
Since we aren't sure about this one, let's put it off for now until we look at it again later.
